napari/plugins/__init__.py

```diff
@@ -268,16 +299,24 @@ def register_function_widget(
     hook_name = '`napari_experimental_provide_function`'
     for func in args if isinstance(args, list) else [args]:
         if not isinstance(func, FunctionType):
-            msg = (
-                f'Plugin {plugin_name!r} provided a non-callable type to '
-                f'{hook_name}: {type(func)!r}. Function widget ignored.'
-            )
+            msg = [
+                trans._(
+                    'Plugin {plugin_name!r} provided a non-callable type to {hook_name}: {func_type!r}. Function widget ignored.',
+                    deferred=True,
+                    plugin_name=plugin_name,
+                    hook_name=hook_name,
+                    func_type=type(func),
+                )
+            ]
             if isinstance(func, tuple):
-                msg += (
-                    " To provide multiple function widgets please use "
-                    "a LIST of callables"
+                msg.append(
+                    trans._(
+                        " To provide multiple function widgets please use a LIST of callables",
+                        deferred=True,
+                    )
                 )
-            warn(msg)
+
+            warn("".join(msg))
```

Comment:
Will that work with deferend ? Wouldn't the "Join" reify the messages ?

Reply:
more specifically is `''.join(...)` a TranslationString or a str ? You may need a custom "join", method.

Reply:
Yeah I might need a custom one. I will do that in a separate PR https://github.com/napari/napari/issues/2599